main.tf

```diff
@@ -313,6 +303,20 @@ resource "aws_codebuild_project" "default" {
     location = var.artifact_location
   }
 
+
+  dynamic "build_batch_config" {
+    for_each = var.concurrent_build_limit == 1 ? [] : [1]
+    content {
+      combine_artifacts = "true"
+      service_role      = join("", aws_iam_role.default.*.arn)
```

Comment:
it's suggested to use a separated role for this one, but i didnt manage to set it up correctly on my own